src/toxcore/dht/server/mod.rs

```diff
@@ -746,6 +748,23 @@ impl Server {
         }
     }
 
+    /// Update returned socket address and time of receiving packet
+    fn update_returned_addr(&self, node: &PackedNode, packet_pk: &PublicKey, close_nodes: &mut Kbucket, friends: &mut Vec<DhtFriend>) {
+        if self.pk == node.pk {
```

Comment:
> A node may have my DHT pk in it's Kbucket. So, my DHT pk may exist in NodesResponse.
Then the node's PK can be updated in my ret_saddr. Is it wrong?

Right, NodesResponse can have our DHT pk but we can't have ourself in our kbucket. See: https://github.com/tox-rs/tox/blob/776a07f6e4d466fec562b1f1c879452baf4684b7/src/toxcore/dht/kbucket.rs#L28-L50

So this if is useless and can be removed.

Reply:
we search Kbucket using packet_pk.

Reply:
ok, you are right. I looked at the wrong key.